doc/Porting.md

```diff
@@ -68,27 +76,37 @@ Implement something like `kernel::hil::UART` for your chip, then submit a pull
 request. Pick a new peripheral and repeat!
 
 
-`board` Crate
--------------
+### `board` Crate
 
 The `board` crate, in `boards/src`, is specific to a physical hardware platform.
 The board file essentially configures the kernel to support the specific
 hardware setup. This includes instantiating drivers for sensors, mapping
 communication buses to those sensors, configuring GPIO pins, etc.
 
-Initializing a board is a bit obtuse and verbose right now, we have some plans
-for how to improve the process, but in the short term the best bet is to start
-from an existing board's `main.rs` file and adapt it. Initially, you will
-likely want to delete most of the capsules and add them slowly as you get
-things working.
+Tock is leveraging "components" for setting up board crates. Components are
+contained structs that include all of the setup code for a particular driver,
+and only require boards to pass in the specific options that are unique to the
+particular platform. For example:
+
+```rust
+let ambient_light = AmbientLightComponent::new(board_kernel, mux_i2c, mux_alarm)
+    .finalize(components::isl29035_component_helper!(sam4l::ast::Ast));
+```
 
-### Board Support
+instantiates the component for an ambient light sensor. Board initiation should
+be largely done using components, but not all components have been created yet,
+so board files are generally a mix of components and verbose driver
+instantiation. The best bet is to start from an existing board's `main.rs` file
+and adapt it. Initially, you will likely want to delete most of the capsules and
+add them slowly as you get things working.
+
```

Comment:
Please add a note that components are singletons -- you can't instantiate them multiple times, and so should all be instantiated in the reset_handler (no recursive instantiation).